MDCStatefulRipple.h

```diff
@@ -0,0 +1,79 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MDCRippleView.h"
+
+/**
+ Provides the current state of the ripple. The ripple is either in its normal state, or in the
+ selected state where the ripple remains spread on the view.
+ - MDCRippleStateNormal: The ripple isn't currently presented.
+ - MDCRippleStateHighlighted: The ripple is activated and shown.
+ - MDCRippleStateSelected: The ripple is in the selected state.
+ - MDCRippleStateDragged: The ripple is in the dragged state.
+ */
+typedef NS_OPTIONS(NSInteger, MDCRippleStateNew) {
+  MDCRippleStateNewNormal = 0,
+  MDCRippleStateNewHighlighted = 1 << 0,
+  MDCRippleStateNewSelected = 1 << 1,
+  MDCRippleStateNewDragged = 1 << 2,
+};
+
+@interface MDCStatefulRippleView : MDCRippleView
+
+/**
+ This BOOL is set to YES if the ripple is currently selected, or NO otherwise.
+ It only has significance if selectionMode is activated.
+ Defaults to NO.
+ */
+@property(nonatomic, getter=isSelected) BOOL selected;
+
+/**
+ This BOOL is set to YES if the ripple is currently highlighted, or NO otherwise.
+ Defaults to NO.
+ */
+@property(nonatomic, getter=isRippleHighlighted) BOOL rippleHighlighted;
+
+/**
+ This BOOL is set to YES if the ripple is currently dragged, or NO otherwise.
+ This state is only triggered manually by setting this property to YES.
+ Defaults to NO.
+ */
+@property(nonatomic, getter=isDragged) BOOL dragged;
+
+/**
+ This BOOL is set to YES if the ripple is currently in the selection mode, or NO otherwise.
+ Defaults to NO.
+ */
+@property(nonatomic) BOOL selectionMode;
+
+/**
+ The current state of the ripple.
+ */
+@property(nonatomic, readonly) MDCRippleStateNew state;
+
+/**
+ Sets the color of the ripple for state.
+ @param rippleColor The ripple color to set the ripple to.
+ @param state The state of the ripple in which to set the ripple color.
+ */
+- (void)setRippleColor:(nullable UIColor *)rippleColor forState:(MDCRippleStateNew)state;
```

Comment:
If components use different states than the state scheme, we will have to constantly convert between state-types in themers.   I'm not sure if there is a better solution, but let's discuss more in the meeting.

Reply:
can you please provide examples on the different states you are referring to?

Reply:
I'm a little confused about the states enums. Is MDCRippleStateNew different than MDCCardControlState? I assume the state subsystem will use something like MDCMaterialControlState, so that will have to be converted to either the ripple or card state enum (note that this might a separate issue which doesn't have to be resolved in this PR).

Reply:
@galiak11 - What is `MDCMaterialControlState`? I can't seem to find it in `develop`.